addons/account_analytic_analysis/account_analytic_analysis.py

```diff
@@ -1,62 +1,37 @@
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+
 from dateutil.relativedelta import relativedelta
 import datetime
 import logging
-import time
+import openerp
 
-from openerp.osv import osv, fields
-import openerp.tools
-from openerp.tools.translate import _
+from openerp import api, fields, models, _
 from openerp.exceptions import UserError
-
 from openerp.addons.decimal_precision import decimal_precision as dp
 
 _logger = logging.getLogger(__name__)
 
-class account_analytic_invoice_line(osv.osv):
+class account_analytic_invoice_line(models.Model):
     _name = "account.analytic.invoice.line"
 
-    def _amount_line(self, cr, uid, ids, prop, unknow_none, unknow_dict, context=None):
-        res = {}
-        for line in self.browse(cr, uid, ids, context=context):
-            res[line.id] = line.quantity * line.price_unit
+    @api.multi
+    @api.depends('quantity', 'price_unit', 'analytic_account_id', 'analytic_account_id.pricelist_id')
+    def _amount_line(self):
+        for line in self:
+            price_subtotal = line.quantity * line.price_unit
             if line.analytic_account_id.pricelist_id:
-                cur = line.analytic_account_id.pricelist_id.currency_id
-                res[line.id] = self.pool.get('res.currency').round(cr, uid, cur, res[line.id])
-        return res
-
-    _columns = {
-        'product_id': fields.many2one('product.product','Product',required=True),
-        'analytic_account_id': fields.many2one('account.analytic.account', 'Analytic Account'),
-        'name': fields.text('Description', required=True),
-        'quantity': fields.float('Quantity', required=True),
-        'uom_id': fields.many2one('product.uom', 'Unit of Measure',required=True),
-        'price_unit': fields.float('Unit Price', required=True),
-        'price_subtotal': fields.function(_amount_line, string='Sub Total', type="float",digits=0),
-    }
-    _defaults = {
-        'quantity' : 1,
-    }
-
+                 price_subtotal = line.analytic_account_id.pricelist_id.currency_id.round(price_subtotal)
+            line.price_subtotal = price_subtotal
+
+    product_id = fields.Many2one('product.product', string='Product', required=True)
+    analytic_account_id = fields.Many2one('account.analytic.account', string='Analytic Account')
+    name = fields.Text(string='Description', required=True)
+    quantity = fields.Float(string='Quantity', required=True, default=1)
+    uom_id = fields.Many2one('product.uom', string='Unit of Measure', required=True)
+    price_unit = fields.Float(string='Unit Price', required=True)
+    price_subtotal = fields.Float(compute='_amount_line', string='Sub Total', digits=0)
+
+    @api.v7
     def product_id_change(self, cr, uid, ids, product, uom_id, qty=0, name='', partner_id=False, price_unit=False, pricelist_id=False, company_id=None, context=None):
```

Comment:
Maybe there's an obvious reason but i don't see why you don't replace the product_id_change() method by the @api.v8 version, and instead have it defined twice!?